File: Data_Structures_and_Algorithms/w4_sorting_and_searching/insertion_sort.c
// insertion sort algorithm in c

#include <stdio.h>
#include "../utils/utils.h"

void insertion_sort(int list[], int size){
	int loc;  //location
	for(int i=0; i < size-1; i++){
		loc = i;
		while(list[loc+1] < list[loc] && loc>-1){
			exchange(&list[loc+1], &list[loc]);
			loc--;
		}
	}
}

int main(){

	int list[] = {9, 7, 6, 15, 16, 5, 10, 11};
	int size = sizeof(list)/sizeof(list[0]);

	printf("Before sorting\n");
	print_list(list, size);
	insertion_sort(list, size);
	printf("\nAfter sorting\n");
	print_list(list, size);

	return 0;
}

File: Data_Structures_and_Algorithms/w4_sorting_and_searching/insertion_sort.py


def insertion_sort(list):
	"""
		list: an array
	"""
	for i in range(len(list)-1):
		loc = i
		while(loc >= 0 and list[loc+1]<list[loc]):
			if list[loc+1] < list[loc]:
				list[loc], list[loc+1]= list[loc+1], list[loc]
			loc-=1


list = [9, 7, 6, 15, 16, 5, 10, 11]

print("Before Sorting")
print(list)
insertion_sort(list)
print("After Sorting")
print(list)